src/main/java/com/solvd/micro9/users/persistence/eventstore/EsUserRepository.java

```diff
@@ -0,0 +1,24 @@
+package com.solvd.micro9.users.persistence.eventstore;
+
+import com.solvd.micro9.users.domain.es.EsStatus;
+import com.solvd.micro9.users.domain.es.EsType;
+import com.solvd.micro9.users.domain.es.EsUser;
+import org.springframework.data.mongodb.repository.Query;
+import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
+import org.springframework.data.repository.query.Param;
+import reactor.core.publisher.Flux;
+
+public interface EsUserRepository extends ReactiveMongoRepository<EsUser, Long> {
+
+    @Query("{'entityId' : :#{#entityId}, 'type' : :#{#type}, 'status' : :#{#status} }")
+    Flux<EsUser> findByEntityIdTypeStatus(@Param("entityId") String entityId,
+                                          @Param("type") EsType type,
+                                          @Param("status") EsStatus status);
+
+    @Query("{$and : [ {entityId : ?0}, {type : ?1}, { $or : [{status: ?2}, {status : ?3}] } ]}")
+    Flux<EsUser> findByEntityIdTypeStatus(@Param("entityId") String entityId,
+                                          @Param("type") EsType type,
+                                          @Param("status") EsStatus status1,
```

Comment:
status1 and status2 is not appropriate naming for params, fix it please

Reply:
@h1alexbel here I'm searching objects by some params including two different statuses, so that an object is either in the 1st status or in the 2nd status. Could you please help me with better namings cause I just have no idea ?

Reply:
@eyakauleva you should check between the status types that you have

Reply:
fixed 4c79ac6